org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/XMLFormatterSettings.java

```diff
@@ -0,0 +1,131 @@
+package org.eclipse.lsp4xml.settings;
+
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Logger;
+
+import javax.swing.text.TabableView;
+
+import org.eclipse.lsp4j.FormattingOptions;
+
+/**
+ * This class is the root of all formatting settings.
+ * It is necessary to update this class for any new additions.
+ * 
+ * All defaults should be set here to eventually be overriden if needed.
+ */
+public class XMLFormatterSettings {
+  private static final Logger LOGGER = Logger.getLogger(XMLFormatterSettings.class.getName());
+
+  //All possible keys
+  public final static String TAB_SIZE = "tabSize";
```

Comment:
Remove TAB_SIZE and INSERT_SPACES constants if you extends from FormattingOptions